backend/src/controllers/datafile.controller.ts

```diff
@@ -38,96 +44,141 @@ export class DatafileController extends Controller {
 
   /**
-   * Retrieves the list of existing files.
+   * Retrieves the list of existing documents.
    *
    * @returns A promise that resolves to an array of Datafile objects.
    */
   @Get()
-  @SuccessResponse(200, "Sent all files.")
+  @SuccessResponse(200, "Sent all documents.")
   public async getAllDataFiles(): Promise<Datafile[]> {
     this.setStatus(200);
     return this.datafileService.getAll();
   }
 
   /**
-   * Retrieves the details of an existing file.
+   * Retrieves the details of an existing document.
    *
-   * @param fileId - The unique identifier of the file.
+   * @param documentId - The unique identifier of the document.
    * @returns A promise that resolves to the Datafile object.
-   * @throws NotFoundError if the file is not found.
+   * @throws NotFoundError if the document is not found.
    */
-  @Get("{fileId}")
+  @Get("{documentId}")
   @Response<NotFoundError>(404, "Not found")
   @SuccessResponse(200, "Datafile found.")
   public async getDatafile(
-    @Path() fileId: MongooseObjectId
+    @Path() documentId: MongooseObjectId
   ): Promise<Datafile> {
     this.setStatus(200);
-    return this.datafileService.get(fileId);
+    return this.datafileService.get(documentId);
   }
 
   /**
-   * Creates a file.
+   * Creates a document.
    *
-   * @param body - The data for creating the file.
+   * @param body - The data for creating the document.
    * @returns A promise that resolves to the created entity.
-   * @throws NotFoundError if the file is not found.
    */
-  @SuccessResponse(201, "Created successfully.") // Custom success response
+  @SuccessResponse(200, "Created successfully.") // Custom success response
   @Post()
   public async createDatafile(
     @Body() body: DatafileCreateParams
   ): Promise<Datafile> {
-    this.setStatus(201); // set return status 201
+    this.setStatus(200);
     return this.datafileService.create(body);
   }
 
   /**
-   * Deletes a file.
+   * Appends the uploaded file to a document with given ID.
    *
-   * @param fileId - The unique identifier of the file to delete.
+   * @param file - The file to append.
+   * @param documentID - The ID of the document to which to append the file
+   * @param fileType - Type of the uploaded file.
+   * @returns A promise that resolves to the updated entity.
+   * @throws OperationNotSupportedError if the file type is not supported.
+   * @throws FailedToParseError if there was an error in parsing the file.
+   * @throws WrongObjectTypeError selected document is not a NOTREFERENCED type.
+   * @throws NotFoundError if the entity is not found.
+   */
+  @SuccessResponse(200, "Appended file successfully.")
+  @Response<NotFoundError>(404, "Not found")
+  @Response<OperationNotSupportedError>(400, "File type not supported.")
+  @Response<FailedToParseError>(400, "Parsing failed.")
+  @Response<WrongObjectTypeError>(400, "Document is not NOTREFERENCED type.")
+  @Post("/file/append")
```

Comment:
i would prefer the id in the route and by a word like attachment to follow rest principles
```suggestion
  @Post("/{documentId}/attachment")
```

Reply:
Done.